webpack/assets/javascripts/react_app/components/Bookmarks/components/BookmarkForm/BookmarkForm.js

```diff
@@ -23,16 +23,18 @@ const BookmarkForm = ({
   controller,
   onCancel,
   initialValues,
+  setModalClosed,
 }) => (
   <ForemanForm
-    onSubmit={(values, actions) =>
-      submitForm({
+    onSubmit={async (values, actions) => {
+      await submitForm({
         url,
         values: { ...values, controller },
         item: 'Bookmarks',
         message: __('Bookmark was successfully created.'),
-      })
-    }
+      });
+      setModalClosed();
+    }}
```

Comment:
wonder if this should be declared in a function elsewhere,
and consumed something like this: `onSubmit={(values, actions) => handleSubmit(values, submitForm)}`

Reply:
Updating to remove inline function